gensrc/thrift/DorisScanService.thrift

```diff
@@ -0,0 +1,162 @@
+}
+
+
+struct TScanColumnData {
+  // One element in the list for every row in the column indicating if there is
+  // a value in the vals list or a null.
+  1: required list<bool> is_null;
+
+  // Only one is set, only non-null values are set. this indicates one column data for a row batch
+  2: optional list<bool> bool_vals;
+  3: optional list<byte> byte_vals;
+  4: optional list<i16> short_vals;
+  5: optional list<i32> int_vals;
+  6: optional list<i64> long_vals;
+  7: optional list<double> double_vals;
+  8: optional list<string> string_vals;
+  9: optional list<binary> binary_vals;
+}
+
+// Serialized batch of rows returned by getNext().
+// one row batch contains mult rows, and the result is arranged in column style
+struct TScanRowBatch {
+  // selected_columns + cols = data frame
+  // Each TScanColumnData contains the data for an entire column. Always set.
+  1: optional list<TScanColumnData> cols
+  2: optional list<TScanColumnDesc> selected_columns
+  // The number of rows returned.
+  3: optional i32 num_rows
+}
+
+// Parameters to open().
+struct TScanOpenParams {
+
+  // tablets to scan
+  1: required list<i64> tablet_ids
+
+  // base64 encoded binary plan fragment
+  2: required string opaqued_query_plan
+
+  // A string specified for the table that is passed to the external data source.
+  // Always set, may be an empty string.
+  3: optional i32 batch_size
+
+  // reserved params for use
+  4: optional map<string,string> payloads
+
+  // The query limit, if specified.
+  5: optional i64 limit
+
+  // The authenticated user name. Always set.
+  // maybe usefullless
+  6: optional string user_name
+}
+
+
+// Returned by open().
+struct TScanOpenResult {
+  1: required TScanStatus status
+  // An opaque context_id used in subsequent getNext()/close() calls. Required.
+  2: optional string context_id
+}
+
+// Parameters to getNext()
+struct TScanNextBatchParams {
+  // The opaque handle returned by the previous open() call. Always set.
+  1: optional string context_id    // doris olap engine context id
+  2: optional i64 offset            // doris should check the offset to prevent duplicate rpc calls
+}
+
+// Returned by getNext().
+struct TScanBatchResult {
+  1: required TScanStatus status
+
+  // If true, reached the end of the result stream; subsequent calls to
+  // getNext() won’t return any more results. Required.
+  2: optional bool eos
+
+  // A batch of rows to return, if any exist. The number of rows in the batch
+  // should be less than or equal to the batch_size specified in TOpenParams.
+  3: optional TScanRowBatch rows
+}
+
+// Parameters to close()
+struct TScanCloseParams {
+  // The opaque handle returned by the previous open() call. Always set.
+  1: optional string context_id
+}
+
+// Returned by close().
+struct TScanCloseResult {
+  1: required TScanStatus status
+}
+
+// scan service expose ability of scanning data ability to other compute system
+service TDorisScanService {
```

Comment:
Can we make this service name more common. If we support load data, we can reuse this service?